pkg/sql/plan/apply_indices_vector.go

```diff
@@ -76,33 +76,62 @@ func (builder *QueryBuilder) applyIndicesForSortUsingVectorIndex(nodeID int32, s
 
 	// 2.b Create Centroids.Version == cast(MetaTable.Version)
 	//     Order By L2 Distance(centroids,	input_literal) ASC limit @probe_limit
-	metaForCurrVersion1, _ := makeMetaTblScanWhereKeyEqVersionAndCastVersion(builder, builder.ctxByNode[nodeID],
+	metaForCurrVersion1, castMetaValueColToBigInt1, _ := makeMetaTblScanWhereKeyEqVersionAndCastVersion(builder, builder.ctxByNode[nodeID],
 		idxTableDefs, idxObjRefs, idxTags, "meta1")
 	centroidsForCurrVersion, _ := makeCentroidsSingleJoinMetaOnCurrVersionOrderByL2DistNormalizeL2(builder,
-		builder.ctxByNode[nodeID], idxTableDefs, idxObjRefs, idxTags, metaForCurrVersion1, distFnExpr, sortDirection)
+		builder.ctxByNode[nodeID], idxTableDefs, idxObjRefs, idxTags, metaForCurrVersion1, distFnExpr, sortDirection, castMetaValueColToBigInt1)
 
 	// 2.c Create Entries.Version ==  cast(MetaTable.Version)
-	metaForCurrVersion2, _ := makeMetaTblScanWhereKeyEqVersionAndCastVersion(builder, builder.ctxByNode[nodeID],
+	metaForCurrVersion2, castMetaValueColToBigInt2, _ := makeMetaTblScanWhereKeyEqVersionAndCastVersion(builder, builder.ctxByNode[nodeID],
 		idxTableDefs, idxObjRefs, idxTags, "meta2")
 	entriesForCurrVersion, _ := makeEntriesCrossJoinMetaOnCurrVersion(builder, builder.ctxByNode[nodeID],
-		idxTableDefs, idxObjRefs, idxTags, metaForCurrVersion2)
+		idxTableDefs, idxObjRefs, idxTags, metaForCurrVersion2, castMetaValueColToBigInt2)
 
 	// 2.d Create JOIN entries and centroids on entries.centroid_id_fk == centroids.centroid_id
+	scanNode.Limit = nil
+	scanNode.Offset = nil
 	entriesJoinCentroids := makeEntriesCrossJoinCentroidsOnCentroidId(builder, builder.ctxByNode[nodeID],
 		idxTableDefs, idxTags,
 		entriesForCurrVersion, centroidsForCurrVersion)
+	//return entriesJoinCentroids
+	// 2.f Sort By l2_distance(vector_col, normalize_l2(literal)) ASC limit original_limit
+	sortTblByL2Distance := makeTblOrderByL2DistNormalizeL2(builder, builder.ctxByNode[nodeID], sortNode,
+		distFnExpr, entriesJoinCentroids, sortDirection, idxTableDefs, idxTags)
 
-	// 2.e Create entries JOIN tbl on entries.original_pk == tbl.pk
 	var pkPos = scanNode.TableDef.Name2ColIndex[scanNode.TableDef.Pkey.PkeyColName] //TODO: watch out.
+
+	onlyUseIndexTables := true
+	if onlyUseIndexTables {
```

Comment:
You need to toggle this to switch between "Index only" result vs "index tables JOIN original table on INDEX join"